### ISSUE TITLE
Test Failure: testInternalRecvStrict (websocket.tests.test_websocket.WebSocketTest) with WebSocketTimeoutException: The read operation timed out

### ISSUE THREAD

**Issue by alunduil** [liris/websocket-client#141]
I'm getting this test failure under pypy with version 0.22.0 of websocket-client:

```
/usr/bin/pypy setup.py test
running test
running egg_info
writing requirements to /var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0-pypy/websocket_client.egg-info/requires.txt
writing /var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0-pypy/websocket_client.egg-info/PKG-INFO
writing top-level names to /var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0-pypy/websocket_client.egg-info/top_level.txt
writing dependency_links to /var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0-pypy/websocket_client.egg-info/dependency_links.txt
reading manifest file '/var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0-pypy/websocket_client.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file '/var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0-pypy/websocket_client.egg-info/SOURCES.txt'
running build_ext
testProxyFromArgs (websocket.tests.test_websocket.ProxyInfoTest) ... ok
testProxyFromEnv (websocket.tests.test_websocket.ProxyInfoTest) ... ok
testSockOpt (websocket.tests.test_websocket.SockOptTest) ... skipped 'Internet-requiring tests are disabled'
testUtf8Validator (websocket.tests.test_websocket.UtilsTest) ... ok
testKeepRunning (websocket.tests.test_websocket.WebSocketAppTest)
A WebSocketApp should keep running as long as its self.keep_running ... skipped 'Internet-requiring tests are disabled'
testSockMaskKey (websocket.tests.test_websocket.WebSocketAppTest)
A WebSocketApp should forward the received mask_key function down ... skipped 'Internet-requiring tests are disabled'
testAfterClose (websocket.tests.test_websocket.WebSocketTest) ... skipped 'Internet-requiring tests are disabled'
testClose (websocket.tests.test_websocket.WebSocketTest) ... ok
testDefaultTimeout (websocket.tests.test_websocket.WebSocketTest) ... ok
testInternalRecvStrict (websocket.tests.test_websocket.WebSocketTest) ... ERROR
testParseUrl (websocket.tests.test_websocket.WebSocketTest) ... ok
testPingPong (websocket.tests.test_websocket.WebSocketTest) ... skipped 'Internet-requiring tests are disabled'
testReadHeader (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecv (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecvContFragmentation (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecvTimeout (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecvWithFireEventOfFragmentation (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecvWithFragmentationAndControlFrame (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecvWithProlongedFragmentation (websocket.tests.test_websocket.WebSocketTest) ... ok
testRecvWithSimpleFragmentation (websocket.tests.test_websocket.WebSocketTest) ... ok
testSecureWebSocket (websocket.tests.test_websocket.WebSocketTest) ... skipped 'Internet-requiring tests are disabled'
testSend (websocket.tests.test_websocket.WebSocketTest) ... ok
testUUID4 (websocket.tests.test_websocket.WebSocketTest)
WebSocket key should be a UUID4. ... ok
testWSKey (websocket.tests.test_websocket.WebSocketTest) ... ok
testWebSocket (websocket.tests.test_websocket.WebSocketTest) ... skipped 'Internet-requiring tests are disabled'
testWebSocketWihtCustomHeader (websocket.tests.test_websocket.WebSocketTest) ... skipped 'Internet-requiring tests are disabled'
testWsUtils (websocket.tests.test_websocket.WebSocketTest) ... No handlers could be found for logger "root"
ok

======================================================================
ERROR: testInternalRecvStrict (websocket.tests.test_websocket.WebSocketTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0/websocket/tests/test_websocket.py", line 265, in testInternalRecvStrict
    data = sock._recv_strict(9)
  File "/var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0/websocket/_core.py", line 949, in _recv_strict
    bytes = self._recv(shortage)
  File "/var/tmp/portage/dev-python/websocket-client-0.22.0/work/websocket-client-0.22.0/websocket/_core.py", line 935, in _recv
    raise WebSocketTimeoutException(message)
WebSocketTimeoutException: The read operation timed out

----------------------------------------------------------------------
Ran 27 tests in 0.040s

FAILED (errors=1, skipped=8)
```

I was wondering if this failure was cosmetic (due to too short of a timeout) or if it indicates a larger problem.  If it does indicate a larger problem I'll continue investigating in my effort to get the ebuild updated for Gentoo with this version of websocket-client.

**Comment by rjschwei** [comment]
I am getting the same failure with openSUSE:Factory and openSUSE:Tumbleweed

https://build.opensuse.org/package/live_build_log/devel:languages:python/python-websocket-client/openSUSE_Factory/x86_64

-> python --version
Python 2.7.9

As I am not familiar with the code I am not certain what information I can provide to help sort this out.

Need some pointers.

**Comment by rjschwei** [comment]
This is not really resolved. THere is still a difference, on some systems where six.PY2 is true the test now fails with:

  File "websocket/tests/test_websocket.py", line 266, in testInternalRecvStrict
     data = sock._recv_strict(9)
     File "websocket/_core.py", line 949, in _recv_strict
     bytes = self._recv(shortage)
   File "websocket/_core.py", line 928, in _recv
     bytes = self.sock.recv(bufsize)
   File "websocket/tests/test_websocket.py", line 57, in recv
     raise e
 SSLError: The read operation timed out

Basically with the patch on a system with python Python 2.7.6 the test fails with the patch (six.PY2 is true) but fails on a system with Python version 2.7.9 (six.PY2 is also true)

**Comment by rjschwei** [comment]
I am not certain if the observed behavior is because of distribution patching or an upstream python change. For the package build and test the following worked:

--- websocket/tests/test_websocket.py.orig
+++ websocket/tests/test_websocket.py
@@ -37,6 +37,17 @@ TEST_WITH_INTERNET = False
 TEST_SECURE_WS = False
 TRACABLE = False
 
+def _versionGreater(targetVer):
+    """Returns true if the interpreter version is greater than the given
+    target version"""
+    interpretVer = sys.version.split()[0]
+    for i in range(len(interpretVer)):
+	if interpretVer[i] == '.':
+	    continue
+        if int(interpretVer[i]) > int(targetVer[i]):
+	    return 1
+
+    return None
 
 def create_mask_key(n):
     return "abcd"
@@ -261,8 +272,12 @@ class WebSocketTest(unittest.TestCase):
         s.add_packet(six.b("baz"))
         with self.assertRaises(ws.WebSocketTimeoutException):
             data = sock._recv_strict(9)
-        with self.assertRaises(SSLError):
-            data = sock._recv_strict(9)
+	if six.PY2 and _versionGreater('2.7.8'):
+	    with self.assertRaises(ws.WebSocketTimeoutException):
+                data = sock._recv_strict(9)
+        else:
+            with self.assertRaises(SSLError):
+                data = sock._recv_strict(9)
         data = sock._recv_strict(9)
         self.assertEqual(data, six.b("foobarbaz"))
         with self.assertRaises(ws.WebSocketConnectionClosedException):

**Comment by liris** [comment]
My python 2.7.6 and 2.7.9 pass the all test case.
We'll skip this case.
